# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/vm/Realm.h
# Commit: e15f1a6f5a38
# Full Hash: e15f1a6f5a38b276216c77f2827945b6f5ec4ea9
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2021-03-06 09:33:16
# Regressor Bug: 1689413
# File Overlap Count: 1
# Description:
#   Bug 1689413 part 9 - Move proto from ObjectGroup to BaseShape. r=jonco
#   
#   The ObjectGroup still stores the proto as well but we don't really use it anymore.
#   This is simpler than removing the group's proto now because that would result in
#   "empty" groups and the group table then doesn't make sense anymore.
# ==============================================================================

diff -r 95fa0ec36f7a -r e15f1a6f5a38 js/src/vm/Realm.h
--- a/js/src/vm/Realm.h	Fri Mar 05 19:10:08 2021 +0000
+++ b/js/src/vm/Realm.h	Fri Mar 05 19:10:08 2021 +0000
@@ -102,7 +102,7 @@
     for (size_t i = 0; i < NumEntries; i++) {
       const Entry& entry = entries_[i];
       if (entry.group && entry.shape->getObjectClass() == clasp &&
-          entry.group->proto() == proto) {
+          entry.group->protoDeprecated() == proto) {
         *group = entry.group;
         *shape = entry.shape;
         return true;